price-discovery/src/redeem_token.rs

```diff
@@ -0,0 +1,133 @@
+elrond_wasm::imports!();
+
+pub const LAUNCHED_TOKEN_REDEEM_NONCE: u64 = 1;
+pub const ACCEPTED_TOKEN_REDEEM_NONCE: u64 = 2;
+
+#[elrond_wasm::module]
+pub trait RedeemTokenModule {
+    #[only_owner]
+    #[payable("EGLD")]
+    #[endpoint(issueRedeemToken)]
+    fn issue_redeem_token(
```

Comment:
please integrate the new function which does everything in ONE go, issue esdt and local roles

Reply:
Done.